src/Stripe.net/Infrastructure/Public/SystemNetHttpClient.cs

```diff
@@ -222,11 +222,23 @@ private string BuildStripeClientUserAgentString()
                 { "bindings_version", StripeConfiguration.StripeNetVersion },
                 { "lang", ".net" },
                 { "publisher", "stripe" },
-                { "lang_version", RuntimeInformation.GetRuntimeVersion() },
-                { "os_version", RuntimeInformation.GetOSVersion() },
                 { "stripe_net_target_framework", StripeNetTargetFramework },
             };
 
+            // The following values are in a try/catch block on the off chance that the
+            // RuntimeInformation methods fail in an unexpected way. This should ~never happen, but
+            // if it does it should not prevent users from sending requests.
+            // See https://github.com/stripe/stripe-dotnet/issues/1986 for context.
+            try
+            {
+                values.Add("lang_version", RuntimeInformation.GetRuntimeVersion());
+                values.Add("os_version", RuntimeInformation.GetOSVersion());
+            }
+            catch (Exception)
+            {
+                // Do nothing.
```

Comment:
Can we add something explicit that we had an error? It would allow us to reliably know how often this happens and how skewed our stats are

Reply:
I think the absence of the `lang_version` and `os_version` keys works just as well as explicit magic values for that.

Reply:
(fixed)